hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsUtils.java

```diff
@@ -776,4 +779,22 @@ public static ContainerProtos.ContainerCommandResponseProto processForDebug(
 
     return msg;
   }
+
+  /**
+   * Redacts sensitive configuration.
+   * Sorts all properties by key name
+   *
+   * @param conf OzoneConfiguration object to be printed.
+   * @return Sorted Map of properties
+   */
+  public static Map<String, String> processForLogging(OzoneConfiguration conf) {
+    Map<String, String> ozoneProps = conf.getOzoneProperties();
+    ConfigRedactor redactor = new ConfigRedactor(conf);
+    Map<String, String> sortedOzoneProps = new TreeMap<>();
+    for (Map.Entry<String, String> entry : ozoneProps.entrySet()) {
+      String value = redactor.redact(entry.getKey(), entry.getValue());
+      sortedOzoneProps.put(entry.getKey(), value);
```

Comment:
This would take care of items with leading/trailing whitespace:

```suggestion
      String value = redactor.redact(entry.getKey(), entry.getValue());
      if (value != null) {
        value = value.trim();
      }
      sortedOzoneProps.put(entry.getKey(), value);
```